docs/usage.md

```diff
@@ -0,0 +1,325 @@
+containing the keys `search` and `replace`. For example:
+
+```yaml
+substitutions:
+  distro_hop:
+    search: 'Red\s?Hat( Enterprise Linux)?'
+    replace: AlmaLinux
+```
+
+That substitution would find anything matching the regular expression,
+including 'RedHat", "Red Hat" and "Red Hat Enterprise Linux", and replace it
+with the string "AlmaLinux" in every file that Stacker writes out, and in every
+dumped file it reads in as well.
+
+This is a powerful feature, but there are some technical details to be aware of
+if you want to use it.
+
+1. These matching patterns are **regular expressions** (*regex*), a language
+   for matching patterns in text. Many common punctuation characters have
+   special meanings in regex, so you will need to prefix them with a backslash
+   (`\`) or they may cause unexpected effects. For example, dots in domain
+   names need to be escaped:
+
+   **WRONG:** `my.domain.net`
+
+   **RIGHT:** `my\.domain\.net`
+
+   If you're unfamiliar with regular expressions, the [Mozilla regular expression
+   cheatsheet](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Regular_expressions/Cheatsheet)
+   describes the special characters in regex and how they affect the search.
+
+1. Stacker's config file is in the [YAML](https://yaml.org/spec/) file format,
+   and YAML handles strings with backslashes in them differently depending on if
+   they're `'single-quoted'` or `"double-quoted"`. In double-quoted strings, the
+   YAML parser will try to interpret the backslash as the part of an escape
+   sequence (for example, it'll replace `\n` with a newline character.)
+   To make sure the YAML parser doesn't mangle your regex, you **MUST** put your
+   `search` string in single quotes.
+
+1. Substitutions are applied to each file in lexicographic order based on the
+   name. This means that as the profile is merged with the defaults, the
+   defaults may be applied before or after the profile. For example:
+
+   ```yaml
+   default:
+     substitutions:
+       01_one:
+       03_three:
+   profiles:
+     staging:
+       substitutions:
+         00_zero:
+         02_two:
+   ```
+
+   In this situation, the substitutions would be applied to your dump files in
+   order, 00-03.
+
+#### `options`
+
+This section holds miscellaneous global options for the Stacker application.
+Currently supported keys are:
+
+- `data_directory`: Where to write out the files dumped from Elasticsearch.
+- `watcher_users`: A mapping of `username: password` pairs for users in the
+  Elastic [Watcher](https://www.elastic.co/guide/en/kibana/current/watcher-ui.html).
+  The Elasticsearch API censors passwords from Watcher users by default, so if
+  you want to load dumped watchers into another cluster you need to provide the
+  password for the Watcher user separately.
+
+#### `log`:
+
+- `log.level`: sets the global log level for the application; accepts values
+  like "INFO" or "warning"
```

Comment:
Does case matter here?

Reply:
UPPER- or lowercase are both acceptable (but mixed-case is not.)
Clarified in f768785a.